.github/workflows/build.yml

```diff
@@ -0,0 +1,48 @@
+name: Build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: "0 0 * * *"
+
+concurrency:
+  group: build-${{ format('{0}-{1}', github.head_ref || github.run_number, github.job) }}
+  cancel-in-progress: true
+
+env:
+  BUILDKIT_PROGRESS: plain
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        ruby_version: [3.3.3, 3.2.4]
+        debian_release: [bullseye, bookworm]
+    timeout-minutes: 360
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
```

Comment:
Can we use the ARM runners provided by GitHub Actions?

Reply:
In theory yes... https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners

But we can't because this repository is public and the above steps rely on logging in to a Docker Hub account for things to work. Therefore, I've decided that this is an OK compromise for now. If we really want to speed things up, we can use a large runner with more cores but I don't think these images will be updated often and even if we do, subsequent changes should be quite simple.